Data/models/Bookmark.swift

```diff
@@ -277,37 +288,67 @@ public final class Bookmark: NSManagedObject, WebsitePresentable, Syncable, CRUD
+    }
+    
+    private class func reorderWithSyncOrder(frc: NSFetchedResultsController<Bookmark>,
+                                            sourceBookmark src: Bookmark,
+                                            destinationBookmark dest: Bookmark,
+                                            sourceIndexPath: IndexPath,
+                                            destinationIndexPath: IndexPath,
+                                            context: NSManagedObjectContext) {
+        
+        guard let srcBgContext = context.object(with: src.objectID) as? Bookmark,
+            let destBgContext = context.object(with: dest.objectID) as? Bookmark else {
             return
         }
         
-        // Warning, this could be a bottleneck, grabs ALL the bookmarks in the current folder
-        // But realistically, with a batch size of 20, and most reads around 1ms, a bottleneck here is an edge case.
-        // Optionally: grab the parent folder, and the on a bg thread iterate the bms and update their order. Seems like overkill.
-        var bms = frc.fetchedObjects!
-        bms.remove(at: bms.index(of: src)!)
-        if sourceIndexPath.row > destinationIndexPath.row {
-            // insert before
-            bms.insert(src, at: bms.index(of: dest)!)
-        } else {
-            let end = bms.index(of: dest)! + 1
-            bms.insert(src, at: end)
-        }
+        let isMovingUp = sourceIndexPath.row > destinationIndexPath.row
         
-        for i in 0..<bms.count {
-            bms[i].order = Int16(i)
+        // Depending on drag direction, all other bookmarks are pushed up or down.
+        if isMovingUp {
+            var prev: String?
+            
+            // Bookmark at the top has no previous bookmark.
+            if destinationIndexPath.row > 0 {
+                let index = IndexPath(row: destinationIndexPath.row - 1, section: destinationIndexPath.section)
+                let objectAtIndexOnBgContext = context.object(with: frc.object(at: index).objectID) as? Bookmark
+                prev = objectAtIndexOnBgContext?.syncOrder
+            }
+            
+            let next = destBgContext.syncOrder
+            srcBgContext.syncOrder = Sync.shared.getBookmarkOrder(previousOrder: prev, nextOrder: next)
+        } else {
+            let prev = destBgContext.syncOrder
+            var next: String?
+            
+            // Bookmark at the bottom has no next bookmark.
+            if let objects = frc.fetchedObjects, destinationIndexPath.row + 1 < objects.count {
+                let index = IndexPath(row: destinationIndexPath.row + 1, section: destinationIndexPath.section)
+                let objectAtIndexOnBgContext = context.object(with: frc.object(at: index).objectID) as? Bookmark
+                next = objectAtIndexOnBgContext?.syncOrder
+            }
+            
+            srcBgContext.syncOrder = Sync.shared.getBookmarkOrder(previousOrder: prev, nextOrder: next)
```

Comment:
Move this outside the `if-else` and place `prev` / `next` as defined outside, and set inside the `if-else`.